rust/core-lib/src/view.rs

```diff
@@ -467,6 +468,38 @@ impl View {
         self.select_region(text, offset, SelRegion::new(start, end), multi_select);
     }
 
+    /// Splits current selections into lines.
+    fn do_split_selection_into_lines(&mut self, text: &Rope) {
+        let mut selection = Selection::new();
+
+        for region in self.selection.iter() {
+            if region.is_caret() {
+                selection.add_region(SelRegion::new(region.max(), region.max()));
```

Comment:
maybe didn't notice this before, but for the record there's a `SelRegion::Caret` fn.